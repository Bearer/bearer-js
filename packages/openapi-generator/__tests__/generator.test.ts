import fs from 'fs'
import path from 'path'

import generator from '../src'

const FUNCTIONS_DIR = path.resolve(path.join(__dirname, '__fixtures__', 'integration', 'functions'))

const functions = fs.readdirSync(FUNCTIONS_DIR).map(func => func.replace(/\.ts$/, ''))

describe('#generator', () => {
  test('generates the valid openapi 3.0 document', () => {
    const integrationUuid = '4l1c3-happy-goats'
    const integrationName = 'happy-goats'
    const document = generator({ functions, integrationUuid, integrationName, functionsDir: FUNCTIONS_DIR })
    expect(document).toMatchSnapshot()
  })
})
