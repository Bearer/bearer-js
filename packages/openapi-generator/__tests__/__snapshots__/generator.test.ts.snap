// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#generator generates the valid openapi 3.0 document 1`] = `
"{
  \\"openapi\\": \\"3.0.0\\",
  \\"info\\": {
    \\"title\\": \\"happy-goats\\"
  },
  \\"servers\\": [
    {
      \\"url\\": \\"https://int.bearer.sh/api/v4/functions/backend/\\"
    }
  ],
  \\"tags\\": [],
  \\"components\\": {
    \\"securitySchemes\\": {
      \\"API Key\\": {
        \\"type\\": \\"apiKey\\",
        \\"in\\": \\"header\\",
        \\"name\\": \\"Authorization\\",
        \\"description\\": \\"You can find your **Bearer API Key** [here](https://app.bearer.sh/keys).\\"
      }
    }
  },
  \\"paths\\": {
    \\"/4l1c3-happy-goats/fetchDataApiKey\\": {
      \\"post\\": {
        \\"parameters\\": [],
        \\"summary\\": \\"fetchDataApiKey\\",
        \\"requestBody\\": {
          \\"content\\": {
            \\"application/json\\": {
              \\"schema\\": {
                \\"type\\": \\"object\\",
                \\"properties\\": {
                  \\"id\\": {
                    \\"type\\": \\"string\\"
                  },
                  \\"name\\": {
                    \\"type\\": \\"string\\"
                  },
                  \\"query\\": {
                    \\"type\\": \\"string\\"
                  }
                }
              }
            }
          }
        },
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"fetchDataApiKey\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"data\\": {
                      \\"type\\": \\"array\\",
                      \\"items\\": {
                        \\"type\\": \\"object\\",
                        \\"properties\\": {
                          \\"id\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"title\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"comments\\": {
                            \\"type\\": \\"object\\",
                            \\"properties\\": {
                              \\"totalCount\\": {
                                \\"type\\": \\"number\\"
                              },
                              \\"visible\\": {
                                \\"type\\": \\"boolean\\"
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          \\"401\\": {
            \\"description\\": \\"Access forbidden\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          },
          \\"403\\": {
            \\"description\\": \\"Unauthorized\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          }
        },
        \\"security\\": [
          {
            \\"API Key\\": []
          }
        ]
      }
    },
    \\"/4l1c3-happy-goats/fetchDataBasic\\": {
      \\"post\\": {
        \\"parameters\\": [],
        \\"summary\\": \\"fetchDataBasic\\",
        \\"requestBody\\": {
          \\"content\\": {
            \\"application/json\\": {
              \\"schema\\": {
                \\"type\\": \\"object\\",
                \\"properties\\": {
                  \\"id\\": {
                    \\"type\\": \\"string\\"
                  },
                  \\"name\\": {
                    \\"type\\": \\"string\\"
                  },
                  \\"query\\": {
                    \\"type\\": \\"string\\"
                  }
                }
              }
            }
          }
        },
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"fetchDataBasic\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"data\\": {
                      \\"type\\": \\"array\\",
                      \\"items\\": {
                        \\"type\\": \\"object\\",
                        \\"properties\\": {
                          \\"id\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"title\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"comments\\": {
                            \\"type\\": \\"object\\",
                            \\"properties\\": {
                              \\"totalCount\\": {
                                \\"type\\": \\"number\\"
                              },
                              \\"visible\\": {
                                \\"type\\": \\"boolean\\"
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          \\"401\\": {
            \\"description\\": \\"Access forbidden\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          },
          \\"403\\": {
            \\"description\\": \\"Unauthorized\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          }
        },
        \\"security\\": [
          {
            \\"API Key\\": []
          }
        ]
      }
    },
    \\"/4l1c3-happy-goats/fetchDataInlineType\\": {
      \\"post\\": {
        \\"parameters\\": [
          {
            \\"name\\": \\"authId\\",
            \\"schema\\": {
              \\"type\\": \\"string\\",
              \\"format\\": \\"uuid\\"
            },
            \\"in\\": \\"query\\",
            \\"description\\": \\"User Identifier\\",
            \\"required\\": true
          }
        ],
        \\"summary\\": \\"fetchDataInlineType\\",
        \\"requestBody\\": {
          \\"content\\": {
            \\"application/json\\": {
              \\"schema\\": {
                \\"type\\": \\"object\\",
                \\"properties\\": {
                  \\"pullRequests\\": {
                    \\"type\\": \\"array\\",
                    \\"items\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          }
        },
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"fetchDataInlineType\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"data\\": {
                      \\"type\\": \\"array\\",
                      \\"items\\": {
                        \\"type\\": \\"object\\",
                        \\"properties\\": {
                          \\"id\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"title\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"comments\\": {
                            \\"type\\": \\"object\\",
                            \\"properties\\": {
                              \\"totalCount\\": {
                                \\"type\\": \\"number\\"
                              },
                              \\"visible\\": {
                                \\"type\\": \\"boolean\\"
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          \\"401\\": {
            \\"description\\": \\"Access forbidden\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          },
          \\"403\\": {
            \\"description\\": \\"Unauthorized\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          }
        },
        \\"security\\": [
          {
            \\"API Key\\": []
          }
        ]
      }
    },
    \\"/4l1c3-happy-goats/fetchDataNone\\": {
      \\"post\\": {
        \\"parameters\\": [],
        \\"summary\\": \\"fetchDataNone\\",
        \\"requestBody\\": {
          \\"content\\": {
            \\"application/json\\": {
              \\"schema\\": {
                \\"type\\": \\"object\\",
                \\"properties\\": {
                  \\"id\\": {
                    \\"type\\": \\"string\\"
                  },
                  \\"name\\": {
                    \\"type\\": \\"string\\"
                  },
                  \\"query\\": {
                    \\"type\\": \\"string\\"
                  }
                }
              }
            }
          }
        },
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"fetchDataNone\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"data\\": {
                      \\"type\\": \\"array\\",
                      \\"items\\": {
                        \\"type\\": \\"object\\",
                        \\"properties\\": {
                          \\"id\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"title\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"comments\\": {
                            \\"type\\": \\"object\\",
                            \\"properties\\": {
                              \\"totalCount\\": {
                                \\"type\\": \\"number\\"
                              },
                              \\"visible\\": {
                                \\"type\\": \\"boolean\\"
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          \\"401\\": {
            \\"description\\": \\"Access forbidden\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          },
          \\"403\\": {
            \\"description\\": \\"Unauthorized\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          }
        },
        \\"security\\": [
          {
            \\"API Key\\": []
          }
        ]
      }
    },
    \\"/4l1c3-happy-goats/fetchDataOauth1\\": {
      \\"post\\": {
        \\"parameters\\": [],
        \\"summary\\": \\"fetchDataOauth1\\",
        \\"requestBody\\": {
          \\"content\\": {
            \\"application/json\\": {
              \\"schema\\": {
                \\"type\\": \\"object\\",
                \\"properties\\": {
                  \\"id\\": {
                    \\"type\\": \\"string\\"
                  },
                  \\"name\\": {
                    \\"type\\": \\"string\\"
                  },
                  \\"query\\": {
                    \\"type\\": \\"string\\"
                  }
                }
              }
            }
          }
        },
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"fetchDataOauth1\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"data\\": {
                      \\"type\\": \\"array\\",
                      \\"items\\": {
                        \\"type\\": \\"object\\",
                        \\"properties\\": {
                          \\"id\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"title\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"comments\\": {
                            \\"type\\": \\"object\\",
                            \\"properties\\": {
                              \\"totalCount\\": {
                                \\"type\\": \\"number\\"
                              },
                              \\"visible\\": {
                                \\"type\\": \\"boolean\\"
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          \\"401\\": {
            \\"description\\": \\"Access forbidden\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          },
          \\"403\\": {
            \\"description\\": \\"Unauthorized\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          }
        },
        \\"security\\": [
          {
            \\"API Key\\": []
          }
        ]
      }
    },
    \\"/4l1c3-happy-goats/fetchDataOauth2\\": {
      \\"post\\": {
        \\"parameters\\": [
          {
            \\"name\\": \\"authId\\",
            \\"schema\\": {
              \\"type\\": \\"string\\",
              \\"format\\": \\"uuid\\"
            },
            \\"in\\": \\"query\\",
            \\"description\\": \\"User Identifier\\",
            \\"required\\": true
          }
        ],
        \\"summary\\": \\"fetchDataOauth2\\",
        \\"requestBody\\": {
          \\"content\\": {
            \\"application/json\\": {
              \\"schema\\": {
                \\"type\\": \\"object\\",
                \\"properties\\": {
                  \\"id\\": {
                    \\"type\\": \\"string\\"
                  },
                  \\"name\\": {
                    \\"type\\": \\"string\\"
                  },
                  \\"query\\": {
                    \\"type\\": \\"string\\"
                  }
                }
              }
            }
          }
        },
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"fetchDataOauth2\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"data\\": {
                      \\"type\\": \\"array\\",
                      \\"items\\": {
                        \\"type\\": \\"object\\",
                        \\"properties\\": {
                          \\"id\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"title\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"comments\\": {
                            \\"type\\": \\"object\\",
                            \\"properties\\": {
                              \\"totalCount\\": {
                                \\"type\\": \\"number\\"
                              },
                              \\"visible\\": {
                                \\"type\\": \\"boolean\\"
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          \\"401\\": {
            \\"description\\": \\"Access forbidden\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          },
          \\"403\\": {
            \\"description\\": \\"Unauthorized\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          }
        },
        \\"security\\": [
          {
            \\"API Key\\": []
          }
        ]
      }
    },
    \\"/4l1c3-happy-goats/fetchDataRefType\\": {
      \\"post\\": {
        \\"parameters\\": [
          {
            \\"name\\": \\"authId\\",
            \\"schema\\": {
              \\"type\\": \\"string\\",
              \\"format\\": \\"uuid\\"
            },
            \\"in\\": \\"query\\",
            \\"description\\": \\"User Identifier\\",
            \\"required\\": true
          }
        ],
        \\"summary\\": \\"fetchDataRefType\\",
        \\"requestBody\\": {
          \\"content\\": {
            \\"application/json\\": {
              \\"schema\\": {
                \\"type\\": \\"object\\",
                \\"properties\\": {
                  \\"reference\\": {
                    \\"type\\": \\"string\\"
                  }
                }
              }
            }
          }
        },
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"fetchDataRefType\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"data\\": {
                      \\"type\\": \\"array\\",
                      \\"items\\": {
                        \\"type\\": \\"object\\",
                        \\"properties\\": {
                          \\"id\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"name\\": {
                            \\"type\\": \\"string\\"
                          },
                          \\"totalCount\\": {
                            \\"type\\": \\"number\\"
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          \\"401\\": {
            \\"description\\": \\"Access forbidden\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          },
          \\"403\\": {
            \\"description\\": \\"Unauthorized\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          }
        },
        \\"security\\": [
          {
            \\"API Key\\": []
          }
        ]
      }
    },
    \\"/4l1c3-happy-goats/fetchDataUndefined\\": {
      \\"post\\": {
        \\"parameters\\": [],
        \\"summary\\": \\"fetchDataUndefined\\",
        \\"requestBody\\": {
          \\"content\\": {
            \\"application/json\\": {}
          }
        },
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"fetchDataUndefined\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {}
                }
              }
            }
          },
          \\"401\\": {
            \\"description\\": \\"Access forbidden\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          },
          \\"403\\": {
            \\"description\\": \\"Unauthorized\\",
            \\"content\\": {
              \\"application/json\\": {
                \\"schema\\": {
                  \\"type\\": \\"object\\",
                  \\"properties\\": {
                    \\"error\\": {
                      \\"type\\": \\"string\\"
                    }
                  }
                }
              }
            }
          }
        },
        \\"security\\": [
          {
            \\"API Key\\": []
          }
        ]
      }
    }
  }
}"
`;
